kernel-rs/src/fs/inode.rs

```diff
@@ -160,31 +298,36 @@ impl InodeGuard<'_> {
         (*dip).minor = inner.minor;
         (*dip).nlink = inner.nlink;
         (*dip).size = inner.size;
-        (*dip).addrs.copy_from_slice(&inner.addrs);
-        kernel().fs().log_write(bp);
+        (*dip).addr_direct.copy_from_slice(&inner.addr_direct);
+        (*dip).addr_indirect = inner.addr_indirect;
+        self.tx.write(bp);
     }
 
     /// Truncate inode (discard contents).
     /// This function is called with Inode's lock is held.
     pub unsafe fn itrunc(&mut self) {
-        for i in 0..NDIRECT {
-            if self.deref_inner().addrs[i] != 0 {
-                bfree(self.dev as i32, self.deref_inner().addrs[i]);
-                self.deref_inner_mut().addrs[i] = 0;
+        let tx = self.tx;
+        let dev = self.dev;
+        for addr in &mut self.deref_inner_mut().addr_direct {
+            if *addr != 0 {
+                tx.bfree(dev, *addr);
+                *addr = 0;
             }
         }
-        if self.deref_inner().addrs[NDIRECT] != 0 {
-            let mut bp = Disk::read(self.dev, self.deref_inner().addrs[NDIRECT]);
+
+        if self.deref_inner().addr_indirect != 0 {
+            let mut bp = Disk::read(self.dev, self.deref_inner().addr_indirect);
```

Comment:
```suggestion
            let mut bp = Disk::read(dev, self.deref_inner().addr_indirect);
```

아래 323번째 줄, 327번째 줄도 동일합니다.